### ISSUE TITLE
"TypeError: must be str, not bytes" while creating deterministic hash

### ISSUE THREAD

**Issue by ParikhKadam** [plasticityai/magnitude#31]
**How to reproduce:**

```
from pymagnitude import *
pos_vectors = FeaturizerMagnitude(100, namespace = "PartsOfSpeech")
print(pos_vectors.query("NN"))
```

**Output:**

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-2f9508aa33b7> in <module>()
      2 pos_vectors = FeaturizerMagnitude(100, namespace = "PartsOfSpeech")
      3 print(pos_vectors.dim) # 4 - number of dims automatically determined by Magnitude from 100
----> 4 print(pos_vectors.query("NN")) # - array([ 0.08040417, -0.71705252,  0.61228951,  0.32322192])
      5 print(pos_vectors.query("JJ")) # - array([-0.11681135,  0.10259253,  0.8841201 , -0.44063763])
      6 print(pos_vectors.query("NN")) # - array([ 0.08040417, -0.71705252,  0.61228951,  0.32322192]) (deterministic hashing so the same value is returned every time for the same key)

C:\Python36\lib\site-packages\pymagnitude\third_party\repoze\lru\__init__.py in cached_wrapper(*args, **kwargs)
    352             else:
    353                 if val is marker:
--> 354                     val = func(*args, **kwargs)
    355                     cache.put(key, val)
    356                 return val

C:\Python36\lib\site-packages\pymagnitude\__init__.py in query(self, q, pad_to_length, pad_left, truncate_left)
    879             vec = self._vector_for_key_cached(q)
    880             if vec is None:
--> 881                 return self._out_of_vocab_vector_cached(q)
    882             else:
    883                 return vec

C:\Python36\lib\site-packages\pymagnitude\third_party\repoze\lru\__init__.py in cached_wrapper(*args, **kwargs)
    352             else:
    353                 if val is marker:
--> 354                     val = func(*args, **kwargs)
    355                     cache.put(key, val)
    356                 return val

C:\Python36\lib\site-packages\pymagnitude\__init__.py in _out_of_vocab_vector_cached(*args, **kwargs)
    334             @lru_cache(None)
    335             def _out_of_vocab_vector_cached(*args, **kwargs):
--> 336                 return self._out_of_vocab_vector(*args, **kwargs)
    337 
    338             @lru_cache(None)

C:\Python36\lib\site-packages\pymagnitude\__init__.py in _out_of_vocab_vector(self, key)
    673             random_vectors = []
    674             for i, ngram in enumerate(ngrams):
--> 675                 seed = self._seed(ngram)
    676                 Magnitude.OOV_RNG_LOCK.acquire()
    677                 np.random.seed(seed=seed)

C:\Python36\lib\site-packages\pymagnitude\__init__.py in _seed(self, val)
    646         """Returns a unique seed for val and the (optional) namespace."""
    647         if self._namespace:
--> 648             return xxhash.xxh32(self._namespace + Magnitude.RARE_CHAR +
    649                                 val.encode('utf-8')).intdigest()
    650         else:

TypeError: must be str, not bytes
```

**Comment by AjayP13** [comment]
Also, confirmed to be a bug (Python 3). Just committed another fix, this one will go out on v0.1.54. Building on the CI/CD now. Thanks!

**Comment by ParikhKadam** [comment]
Thank you.. I am waiting for the newer version.

**Comment by AjayP13** [comment]
The fix is out on 0.1.55. You can do pip install -U pymagnitude (Python 2.7) or pip3 install -U pymagnitude (Python 3) to upgrade.
